doc/md/language-manual.md

```diff
@@ -1654,6 +1654,23 @@ Such an object literal, sometimes called a *record*, is equivalent to the object
 
 Object expressions support *punning* for concision. A punned field `<id>` is shorthand for `<id> = <id>`; Similarly, a typed, punned field `<id> : <typ>` is short-hand for `<id> = <id> : <typ>`. Both associate the field named `<id>` with the value of the identifier `<id>`.
 
+#### Object reuse and extension
+
+Objects can be combined and/or extended using the `and` and `with` keywords.
+
+A record expression `{ <exp> (and <exp>)* (with <exp-field>;+)? }` merges the objects (or modules) specified as *base* expressions, and augments the result to also contain the specified fields. The `with <exp-field>;+` clause can be omitted when at least two bases appear and none have common field labels.
+Thus the field list serves as
+- disambiguation for field labels occurring more than once in the bases,
+- supplying fresh fields,
+- changing field types, and
+- adding new `var` field or to avoid aliasing with an existing `var` field from some base.
+
+The resulting type is determined by the bases' (and explicitly given fields') static type.
+
+`var` fields from the bases (when not disambiguated) will show up in the result as `var` fields that *share* the original contents,
+i.e. an assignment to such a field of either the base or the result will change the shared content. If such a behaviour is not desired,
+then the field must be explicitly overwritten in the explicit field list.
```

Comment:
Any `var` field from some base must be overwritten in the explicit field list, to avoid introducing aliasing of var fields.